### PR TITLE
feat(update): #MC-234 create API to get calendar by id

### PR DESCRIPTION
**Pull request by alicedraillard**
## Describe your changes
allow user to make requests to get his own calendars

## Checklist tests
call API with postman and valid calendar id

## Issue ticket number and link
[MC-234](https://jira.support-ent.fr/browse/MC-234)

## Checklist before requesting a review (magic string, indentation, comment/documentation...)

- [x] I have detailed the tests to do in my feature/fix in order to prevent consequents regressions (must specify in **Checklist tests**)
- [x] I have performed a self-review of my code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation (API Doc etc...) - (must specify in **Description** for target version)
- [ ] If it is a consequent feature, I have added thorough tests.
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] Any dependent changes have been added to this project (must specify in **Description**)

### CODE REVIEW

File: src/main/java/net/atos/entng/calendar/controllers/CalendarController.java [hunk 1]
```diff
@@ -117,6 +117,33 @@ public void listCalendars(HttpServerRequest request) {
...
+    @Get("/:id")
+    @SecuredAction("calendar.get")
+    @Trace(Actions.GET_CALENDAR)
+    public void getCalendar(HttpServerRequest request) {
+        UserUtils.getUserInfos(eb, request, user -> {
+            if (user == null) {
+                unauthorized(request);
+                return;
+            }
+            final String id = request.params().get("id");
```
Comment (flmartineau) [inline src/main/java/net/atos/entng/calendar/controllers/CalendarController.java:129]: Field.ID

File: src/main/java/net/atos/entng/calendar/controllers/CalendarController.java [hunk 2]
```diff
@@ -117,6 +117,33 @@ public void listCalendars(HttpServerRequest request) {
         list(request);
     }
 
+    @Get("/:id")
+    @SecuredAction("calendar.get")
```
Comment (flmartineau) [inline src/main/java/net/atos/entng/calendar/controllers/CalendarController.java:121]: Vous avez pas d'enum pour stocker les SecuredActions ?

File: src/main/java/net/atos/entng/calendar/controllers/CalendarController.java [hunk 3]
```diff
@@ -117,6 +117,33 @@ public void listCalendars(HttpServerRequest request) {
...
+            if (id == null || id.trim().isEmpty()) {
+                badRequest(request);
+                return;
+            }
+            calendarService.list(Collections.singletonList(id), true, user.getUserId())
+                    .onSuccess(result -> {
+                        Renders.renderJson(request, new JsonObject().put(Field.CALENDAR, result));
+                    })
+                    .onFailure(error -> {
+                        String message = String.format("[Calendar@%s::listCalendar] An error has occured" +
```
Comment (flmartineau) [inline src/main/java/net/atos/entng/calendar/controllers/CalendarController.java:139]: [Calendar@%s::getCalendar]

File: src/main/java/net/atos/entng/calendar/core/constants/Field.java [hunk 4]
```diff
@@ -144,4 +144,7 @@ private Field() {
     public static final String POST = "post";
     public static final String PUT = "put";
 
+    //Mongo
+    public static final String OWNERUSERID = "owner.userId";
```
Comment (flmartineau) [inline src/main/java/net/atos/entng/calendar/core/constants/Field.java:148]: C'est ok mais une autre méthode est de déclarer deux Field : OWNER et USERID et ensuite dans ton code tu fais 
`String.format("%s.%s", Field.OWNERID, Field.USERID)`.
C'est un peu long mais si jamais tu as plusieurs combinaisons ca t'évite de créer autant de constantes Field.

**Review by flmartineau — APPROVED** [review]

**Review by sjollois — APPROVED** [review]

**Review by qperie — APPROVED** [review]

**Pull request merged**